lib/config/config.js

```diff
@@ -103,6 +107,25 @@ class Config {
       gatewayWatcher.isReady = true;
     });
   }
+
+  // this function can preserve comments and formatting.
+  // due to fragile nature of YAWN it may not work properly in that mode.
+  // safe way is to use it in rewrite file mode
+  update (modifier) {
+    return readFile(this.gatewayConfigPath, 'utf8').then(data => {
+      let text;
+      if (this.systemConfig && this.systemConfig.preserveStructureOnUpdates) {
```

Comment:
@kevinswiber seems to be a compromise on yawn with comments vs nuke it all approach? 
by default it will be nuke it all, and if people complain it will be a doc section describing this flag
(obviously name can be better)